src/core/inlines.js

```diff
@@ -22,6 +22,71 @@ import { renderInlineCitation } from "./render-biblio";
 export const name = "core/inlines";
 export const rfc2119Usage = {};
 
+/**
+ * @return {HTMLElement}
+ * @param {string} matched
+ */
+function inlineRFC2119Matches(matched) {
+  const normalize = matched.split(/\s+/).join(" ");
+  const nodeElement = hyperHTML`<em class="rfc2119" title="${normalize}">${normalize}</em>`;
+  // remember which ones were used
+  rfc2119Usage[normalize] = true;
+  return nodeElement;
+}
+
+/**
+ * @param {string} matched
+ */
+function inlineXrefMatches(matched) {
+  // slices "{{{" at the beginning and "}}}" at the end
+  const ref = matched.slice(3, -3).trim();
+  return ref.startsWith("\\")
+    ? document.createTextNode(`{{{${ref.slice(1)}}}}`)
+    : idlStringToHtml(ref);
+}
+
+/**
+ * @return {Iterable<Node>}
+ * @param {string} matched
+ * @param {Text} txt
+ * @param {Object} conf
+ */
+function inlineBibrefMatches(matched, txt, conf) {
+  // slices "[[" at the start and "]]" at the end
+  const ref = matched.slice(2, -2);
+  if (ref.startsWith("\\")) {
+    return [document.createTextNode(`[[${ref.slice(1)}]]`)];
+  }
+  const { type, illegal } = refTypeFromContext(ref, txt.parentNode);
+  const cite = renderInlineCitation(ref);
+  const cleanRef = ref.replace(/^(!|\?)/, "");
+  if (illegal && !conf.normativeReferences.has(cleanRef)) {
+    showInlineWarning(
+      cite.childNodes[1], // cite element
+      "Normative references in informative sections are not allowed. " +
+        `Remove '!' from the start of the reference \`[[!${ref}]]\``
+    );
+  }
+
+  if (type === "informative" && !illegal) {
+    conf.informativeReferences.add(cleanRef);
+  } else {
+    conf.normativeReferences.add(cleanRef);
+  }
+  return cite.childNodes;
+}
+
+/**
+ * @param {string} matched
```

Comment:
nit: missing `@return`

Reply:
well @saschanaz had suggested that the type was obvious so I can just leave that out.

Reply:
Yes, it can be inferred automatically by the compiler. It can still be manually written but it adds maintenance burden.

Reply:
Hmm... ok yeah. I'm kinda 50/50 on this, as I like the code comments primarily for human usage: They help me outside the context of VS Code - like when doing code review or navigating the code through GitHub's web interface.

Reply:
so what should I do :sweat_smile: ? shall I add those return types?

Reply:
>like when doing code review

I think reviewing types should be automatically done by typescript. That's why we use it, right?

Reply:
Agree.